File: src/ast/CtorVtableSpecialName.java
package ast;

import java.io.StringWriter;

public class CtorVtableSpecialName extends BaseNode {
    private BaseNode _firstType;
    private BaseNode _secondType;

    public CtorVtableSpecialName(BaseNode firstType, BaseNode secondType) {
        super(NodeType.CtorVtableSpecialName);
        _firstType = firstType;
        _secondType = secondType;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write("construction vtable for ");
        _firstType.Print(writer);
        writer.write("-in-");
        _secondType.Print(writer);
    }
}

File: src/de/fearlesstobi/demangler/ast/CtorVtableSpecialName.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class CtorVtableSpecialName(private val firstType: BaseNode?, private val secondType: BaseNode) : BaseNode(NodeType.CtorVtableSpecialName) {
    public override fun printLeft(writer: StringWriter) {
        writer.write("construction vtable for ")
        firstType!!.print(writer)
        writer.write("-in-")
        secondType.print(writer)
    }

}